pkg/asset/templates/kubelet.yaml

```diff
@@ -24,20 +24,60 @@ spec:
         - --
         - ./hyperkube
         - kubelet
-        - --api-servers={{ index .APIServers 0 }}
-        - --config=/etc/kubernetes/manifests
-        - --allow-privileged
+        - --api-servers=$(API_SERVERS)
+        - --config=$(CONFIG)
+        - --allow-privileged=$(ALLOW_PRIVILEGED)
         - --hostname-override=$(MY_POD_IP)
-        - --cluster-dns=10.3.0.10
-        - --cluster-domain=cluster.local
-        - --kubeconfig=/etc/kubernetes/kubeconfig
-        - --lock-file=/var/run/lock/kubelet.lock
-        - --minimum-container-ttl-duration=3m0s
+        - --cluster-dns=$(CLUSTER_DNS)
+        - --cluster-domain=$(CLUSTER_DOMAIN)
+        - --kubeconfig=$(KUBECONFIG)
+        - --lock-file=$(LOCK_FILE)
+        - --minimum-container-ttl-duration=$(MINIMUM_CONTAINER_TTL_DURATION)
         env:
           - name: MY_POD_IP
             valueFrom:
               fieldRef:
                 fieldPath: status.podIP
+          - name: API_SERVERS
```

Comment:
Sure that's fine. Later we may want to still use volumes for the config as it might give us more flexibility (e.g. you can modify a volume during runtime, but you can't modify env vars or the pod spec). Not a huge deal, and this will rely on the upstream component config soon - so we'll be changing it again either way.